pilot/pkg/proxy/envoy/config.go

```diff
@@ -270,8 +270,8 @@ func buildSidecarListenersClusters(
 		// only HTTP outbound clusters are needed
 		httpOutbound := buildOutboundHTTPRoutes(mesh, node, instances, services, config)
 		httpOutbound = buildEgressHTTPRoutes(mesh, node, instances, config, httpOutbound)
-		clusters = append(clusters,
-			httpOutbound.clusters()...)
+		httpOutbound = buildExternalServiceHTTPRoutes(mesh, node, instances, config, httpOutbound)
+		clusters = append(clusters, httpOutbound.clusters()...)
```

Comment:
this is added only for the proxy port listener.. what about the normal one?

Reply:
it is called in `buildOutboundListeners` here:
https://github.com/istio/istio/pull/2811/files/5e652dd81769d3cca81e9c9d66f71446e781b5a9#diff-15b4be428ba285dd7fb68ad5b8e35c8aR505